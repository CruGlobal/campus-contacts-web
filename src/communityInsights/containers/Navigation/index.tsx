import React, { useContext } from 'react';
import styled from '@emotion/styled';
import { NavLink } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import { AppContext } from '../../../appContext';

const NavigationBar = styled.div`
    background: #005a7f;
    height: 52px;
    padding-top: 15px;
    padding-bottom: 15px;
    padding-left: 18px;
    display: flex;
    justify-content: flex-start;
    border-radius: 0 0 8px 8px;
`;

const Link = styled(NavLink)`
    font-family: Titillium Web;
    font-style: normal;
    font-weight: normal;
    line-height: 21px;
    margin-right: 20px;
    color: white;
    position: relative;

    :hover {
        color: ${({ theme }) => theme.colors.highlight};
        text-decoration: none;
    }

    &.active {
        color: ${({ theme }) => theme.colors.highlight};

        ::after {
            content: '';
            position: absolute;
            bottom: -15px;
            left: 40%;
            border-bottom: 6px solid #f2f2f2;
            border-left: 6px solid transparent;
            border-right: 6px solid transparent;
            width: 0;
        }
    }
`;

const Navigation = () => {
    const { orgId } = useContext(AppContext);
    const { t } = useTranslation('insights');
    return (
        <NavigationBar>
            <Link to={`/ministries/${orgId}/insights/personal`}>
                {t('tabs.personal')}
            </Link>
            <Link to={`/ministries/${orgId}/insights/steps`}>
                {t('tabs.steps')}
            </Link>
            <Link to={`/ministries/${orgId}/insights/interactions`}>
                {t('tabs.interactions')}
            </Link>
            <Link to={`/ministries/${orgId}/insights/challenges`}>
                {t('tabs.challenges')}
            </Link>
        </NavigationBar>
    );
};

export default Navigation;
