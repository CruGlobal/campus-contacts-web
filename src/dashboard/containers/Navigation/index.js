// Vendors
import React from 'react';
import styled from '@emotion/styled';
import { NavLink } from 'react-router-dom';
// Project
import AppContext from '../../appContext';

const NavigationBar = styled.div`
    background: #007398;
    height: 52px;
    padding-top: 15px;
    padding-bottom: 15px;
    padding-left: 18px;
    display: flex;
    justify-content: flex-start;
`;

const Link = styled(NavLink)`
    font-family: Titillium Web;
    font-style: normal;
    font-weight: normal;
    line-height: 21px;
    margin-right: 20px;
    color: white;
    position: relative;

    :hover {
        color: ${props => props.theme.colors.highlight};
        text-decoration: none;
    }

    &.active {
        color: ${props => props.theme.colors.highlight};
    }
`;

const Navigation = () => {
    const { orgId } = React.useContext(AppContext);
    return (
        <NavigationBar>
            <Link to={`/ministries/${orgId}/insights/personal`}>
                Personal Steps
            </Link>
            <Link to={`/ministries/${orgId}/insights/steps`}>
                Steps of Faith
            </Link>
            <Link to={`/ministries/${orgId}/insights/interactions`}>
                Interactions
            </Link>
            <Link to={`/ministries/${orgId}/insights/challenges`}>
                Challenges
            </Link>
        </NavigationBar>
    );
};

export default Navigation;

Navigation.propTypes = {};
