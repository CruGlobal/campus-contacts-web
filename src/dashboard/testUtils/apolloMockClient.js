import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
