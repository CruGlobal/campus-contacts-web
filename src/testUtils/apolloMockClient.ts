import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
