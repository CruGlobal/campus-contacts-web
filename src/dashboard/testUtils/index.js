import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
