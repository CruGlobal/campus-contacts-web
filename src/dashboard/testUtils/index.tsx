import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
