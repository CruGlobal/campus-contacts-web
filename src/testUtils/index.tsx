import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
