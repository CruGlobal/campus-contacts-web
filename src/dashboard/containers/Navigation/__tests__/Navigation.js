// Vendors
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
// Project
import { renderWithContext } from '../../../testUtils';
// Subject
import Navigation from '../';

describe('<Navigation />', () => {
    it('should render properly', async () => {
        renderWithContext(
            <Router>
                <Navigation />
            </Router>,
            {},
            { orgId: 1 },
        ).snapshot();
    });
});
