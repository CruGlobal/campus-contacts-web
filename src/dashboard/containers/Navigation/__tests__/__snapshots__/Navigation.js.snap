// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render properly 1`] = `
<div>
  <div
    class="css-1qdtt6o"
  >
    <a
      class="css-cdd0f3"
      href="/ministries/1/insights/personal"
    >
      Personal Steps
    </a>
    <a
      class="css-cdd0f3"
      href="/ministries/1/insights/steps"
    >
      Steps of Faith
    </a>
    <a
      class="css-cdd0f3"
      href="/ministries/1/insights/interactions"
    >
      Interactions
    </a>
    <a
      class="css-cdd0f3"
      href="/ministries/1/insights/challenges"
    >
      Challenges
    </a>
  </div>
</div>
`;
